### PR TITLE
Added code for RubrikPolaris Release 1.2.0

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description

Dependent PR: https://github.com/demisto/dockerfiles/pull/13591
##### Rubrik Radar

- Updated the Docker image to: *demisto/rubrik-polaris-sdk-py3:1.0.0.50158*.
- Updated the rubrik-event-list command's parameters possible values.
- Updated the rubrik-polaris-vm-object-snapshot-list command's possible values.
- Updated the rubrik-polaris-object-list command to add type_filter argument as required parameter.
- Updated the rubrik-gps-vm-livemount command and used new function with updated parameters provided by the client.
- Updated the **Rubrik Data Object Discovery - Rubrik Polaris** playbook and attached sub playbook to it.
- Added the **Rubrik List Snapshots - Rubrik Polaris** sub playbook.

##### Rubrik List Snapshots - Rubrik Polaris
- Added a playbook which will list snapshots for all objects. (Available from Cortex XSOAR 6.0.0).

##### Rubrik Data Object Discovery - Rubrik Polaris

- Updated the playbook and attached **Rubrik List Snapshots - Rubrik Polaris** sub playbook for Data discovery of the object available in the incident. (Available from Cortex XSOAR 6.0.0).

##### RubrikCDMClusterConnectionState

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### DateToTimeStamp

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### RubrikRadarFilesModified

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### RubrikSonarSensitiveHits

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### RubrikRadarFilesAdded

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### RubrikSonarOpenAccessFiles

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### RubrikRadarFilesDeleted

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.
##### RubrikSonarTotalHits

- Updated the Docker image to: *demisto/python3:3.10.10.50695*.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
-// Development container for Content. Poetry hash 1.11.0
+// Development container for Content. Poetry hash 6c69ac4b4e71cfdddd29602e19df186c1118fb8ad974df7b2091dd9417b7aec9
```
Comment (yucohen) [inline .devcontainer/devcontainer.json:1]: please revert

File: Packs/RubrikPolaris/Integrations/RubrikPolaris/test_data/fetch_incidents_success_incidents.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
...
...r15'.\", \"severity\": \"Info\", \"time\": \"2021-10-22T15:56:45.640Z\"}, {\"id\": \"40604560-cc5e-4756-b83d-1070ea2d1d75\", \"message\": \"Detected significant indication of encrypted files.\", \"severity\": \"Warning\", \"time\": \"2021-10-22T15:56:30.665Z\"}, {\"id\": \"86534887-0b64-4603-bddd-aacf323a4adf\", \"message\": \"Starting ransomware detection job on cluster sand1-rbk01 for snapshot taken on 2021-10-22 13:22:25.18 +0000 UTC of VmwareVirtualMachine 'sx1-radar15'.\", \"severity\": \"Info\", \"time\": \"2021-10-22T15:52:13.698Z\"}, {\"id\": \"426fee25-3ba9-4865-acb2-c4fafd9c8705\", \"message\": \"Started detailed encryption analysis for snapshot taken on 2021-10-22 13:22:25.18 +0000 UTC of VmwareVirtualMachine 'sx1-radar15'.\", \"severity\": \"Info\", \"time\": \"2021-10-22T15:52:12.454Z\"}, {\"id\": \"629a828f-dbbd-42ca-995f-6a96e3a47a23\", \"message\": \"Started Radar analysis for snapshot taken on 2021-10-22 13:22:25.18 +0000 UTC of VmwareVirtualMachine 'sx1-radar15'.\", \"severity\": \"Info\", \"time\": \"2021-10-22T15:52:11.875Z\"}]}, \"eventCompleted\": \"True\", \"radar_files_added\": \"3492\", \"radar_files_modified\": \"136\", \"radar_files_deleted\": \"3395\"}",
```
Comment (yucohen) [inline Packs/RubrikPolaris/Integrations/RubrikPolaris/test_data/fetch_incidents_success_incidents.json:11]: Do we use this rawJSON in the tests? If not, lets delete this from the JSON

**Review by yucohen — CHANGES_REQUESTED** [review]
@crestdatasystems Thanks for your contribution! We highly appreciate it
Please see my comments

File: Packs/RubrikPolaris/Integrations/RubrikPolaris/RubrikPolaris.yml [hunk 3]
```diff
@@ -1695,105 +1695,130 @@ script:
       type: String
   - arguments:
     - default: false
-      description: "The ID of the snapshot that is to be mounted.\n\nNote: Users\
-        \ can get the snapshot ID by executing the \"rubrik-polaris-vm-object-snapshot-list\"\
-        \ command."
+      description: The snappable ID.
       isArray: false
-      name: snapshot_id
+      name: snappable_id
```
[the remark this answers is not in the document]
Comment (jyotipatel2090) [reply Packs/RubrikPolaris/Integrations/RubrikPolaris/RubrikPolaris.yml:1700]: Yes, we want to change this for the existing users also this is not working.

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -83,26 +83,26 @@
...
+	// 	},
+	// 	"ghcr.io/devcontainers/features/node:1": {
+	// 		"version": "latest"
+	// 	}
+	// },
+	// "overrideFeatureInstallOrder": [
+	// 	"ghcr.io/devcontainers/features/common-utils:1"
+	// ],
+	"onCreateCommand": "dos2unix -n .devcontainer/createCommand.sh .devcontainer/createCommand_unix.sh && chmod +x .devcontainer/createCommand_unix.sh && bash .devcontainer/createCommand_unix.sh"
 }
```
Comment (yucohen) [inline .devcontainer/devcontainer.json:108]: please revert those as well

**Comment by edik24** [comment]
@jyotipatel2090 please fix all docker validation errors

**Comment by jyotipatel2090** [comment]
> @jyotipatel2090 please fix all docker validation errors

Fixed, Except one which is related to backward compatibility

**Review by yucohen — APPROVED** [review]

**Pull request merged**